### PR TITLE
UX-622 Convert Drawer to typescript

### PR DESCRIPTION
**Pull request by jonambas**
### What Changed
- Converts `Drawer` to typescript
- Updates GH action ubuntu version

### How To Test or Verify
- Verify Drawer tests and stories work as expected

### PR Checklist

- [x] Add the correct `type` label
- [ ] Pull request approval from #uxfe or #design-guild

### CODE REVIEW

File: packages/matchbox/src/helpers/window.ts [hunk 1]
```diff
@@ -22,6 +24,8 @@ export function getWindow(): GetWindowReturnType {
     matchMedia: noop,
     addEventListener: noop,
     removeEventListener: noop,
+    setTimeout: noop,
+    clearTimeout: noop,
```
Comment (jonambas) [inline packages/matchbox/src/helpers/window.ts:28]: There has to be a better way to do this

File: packages/matchbox/src/components/Drawer/Drawer.tsx [hunk 2]
```diff
@@ -17,7 +16,19 @@ import { getWindow, isInIframe } from '../../helpers/window';
...
 import { Overlay, Container } from './styles';
 
-const Drawer = React.forwardRef(function Drawer(props, userRef) {
+export type DrawerProps = {
+  children?: React.ReactNode;
+  closeOnEscape?: boolean;
+  closeOnOutsideClick?: boolean;
+  id: string;
+  onChange: (open?: boolean) => void;
+  onClose: () => void;
```
Comment (logansparlin) [inline packages/matchbox/src/components/Drawer/Drawer.tsx:25]: should these be optional?

File: packages/matchbox/src/components/Drawer/Content.tsx [hunk 3]
```diff
@@ -1,15 +1,17 @@
...
-import { createPropTypes } from '@styled-system/prop-types';
+import { padding, PaddingProps } from 'styled-system';
 import { pick } from '@styled-system/props';
 
 const Container = styled.div`
   ${padding}
 `;
 
-const Content = React.forwardRef(function Content(props, ref) {
+type DrawerContentProps = PaddingProps & {
```
Comment (logansparlin) [inline packages/matchbox/src/components/Drawer/Content.tsx:10]: Should export these props since we're exporting the props from the other files

**Review by logansparlin — APPROVED** [review]
looks good, jus ta couple small comments

**Pull request merged**